apps/emqx_durable_storage/src/emqx_ds.erl

```diff
@@ -77,16 +77,14 @@
 
 -type iterator_id() :: binary().
 
-%%-type session() :: #session{}.
-
 -type message_store_opts() :: #{}.
 
 -type message_stats() :: #{}.
 
 -type message_id() :: binary().
 
 %% Parsed topic:
--type topic() :: list(binary()).
+-type topic() :: list(binary() | '+' | '#' | '').
```

Comment:
Hmm, I don't think this is correct: by the spec the topic cannot include `+` or `#` levels. This change essentially turned this type into a `topic_filter()` type, equivalent to `-type words() :: list(word()).` from `emqx_types`